traits/tests/test_regression.py

```diff
@@ -309,6 +310,18 @@ class A(HasTraits):
         with self.assertRaises(ZeroDivisionError):
             a.bar = "foo"
 
+    def test_clone_list_of_enum_trait(self):
+        # Regression test for enthought/traits#1622.
+
+        class Company(HasTraits):
+            departments = List(Str)
+            projects = List(Enum(values="departments"))
+
+        company = Company(departments=["sales"], projects=["sales"])
+        clone = company.clone_traits()
+
+        self.assertEqual(clone.projects, ["sales"])
+
```

Comment:
Would it be worth testing that the clone only allows to select among the given choices? Testing that the clone attributes are correctly independent from the original object traits?
```suggestion

        order.selection.append('fish')
        self.assertEqual(clone.selection, ['fish'])

        with self.assertRaises(TraitError):
            clone.selection.append("bouillabaisse")
```

Reply:
> Would it be worth testing that the clone only allows to select among the given choices?

That's unfortunately the part that's not fixed by this PR, and that's much harder to fix in general. Currently in the cloning operation the `TraitListObject` becomes disconnected from the original `HasTraits` object - that means that it continues to function normally as a list (you can append to it, etc.), but the validation can't work since that validation depends on the object.

Reply:
> That's unfortunately the part that's not fixed by this PR

Let me test that statement by merging your suggestion. :-) If I'm right, the CI should fail, and then we can revert ...